provisioning/device/samples/solutions/BestPracticeSampleX509/Cleanup.ps1

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+param(
+    # Resource group name of the IoT dps.
+    [Parameter(Mandatory=$true)]
+    [string] $resourceGroup,
+
+    # IoT dps name used for running the sample.
+    [Parameter(Mandatory=$true)]
+    [string] $dpsName,
+
+    # IoT hub name used for running the sample.
+    [Parameter(Mandatory=$true)]
+    [string] $iothubName,
+
+    # Device Id created to use in sample.
+    [Parameter(Mandatory=$true)]
+    [string] $deviceId,
+    
+    [Parameter(Mandatory=$false)]
+    [string] $groupEnrollmentId = "x509GroupEnrollment"
+)
+
+# Check if the resource group exists. If not, exit.
+$resourceGroupExists = az group exists -n $resourceGroup
+if ($resourceGroupExists -ne $true){
+    Write-Host "Resource Group '$resourceGroup' does not exist. Exiting..."
+    exit
+}
+
+# Check if the DPS instance exists. If not, exit.
+$dpsExists = az iot dps show --name $dpsName -g $resourceGroup 2>nul
+if ($dpsExists -eq $null){
+    Write-Host "DPS '$dpsName' does not exist under '$resourceGroup'. Exiting..."
```

Comment:
Why exit? Could be the other resources are still there.

Reply:
This is to check if the dps instance exists before deleting the enrollment group created earlier.  If there is no dps instance, then no enrollment group to delete.

Reply:
Instead of exiting here, I can skip deleting the enrollment group if the dps instance doesn't exist and then just delete the device (if it exists).

Reply:
I mean, maybe change the logic from not exists to exists, and delete inside the braces if the things you need are there?

Reply:
Makes sense! Thank you for the suggestion and the comments!